packages/mapviewer/tests/cypress/tests-e2e/importToolFile.cy.js

```diff
@@ -563,61 +583,98 @@ describe('The Import File Tool', () => {
-                if (index === 0) {
-                    cy.get(`[data-cy^="floating-button-has-error-${onlineUrlNotReachable}"]`)
+
+                const errorData = [
+                    {
+                        url: validOnlineUrlWithInvalidContentType,
+                        shouldHaveError: false,
+                    },
+                    {
+                        url: onlineUrlNotReachable,
+                        shouldHaveError: true,
+                        errorMessage: 'file not accessible',
+                    },
+                    {
+                        url: invalidFileOnlineUrl,
+                        shouldHaveError: true,
+                        errorMessage: 'Invalid file',
+                    },
+                    {
+                        url: outOfBoundKMLUrl,
+                        shouldHaveError: true,
+                        errorMessage: 'out of projection bounds',
+                    },
+                ]
+
+                const { url, shouldHaveError, errorMessage } = errorData[index]
+
+                if (shouldHaveError) {
+                    cy.wrap($layer)
+                        .find('[data-cy^="button-has-error"]')
                         .should('be.visible')
-                        .contains('file not accessible')
-                } else if (index === 1) {
-                    cy.get(`[data-cy^="floating-button-has-error-${invalidFileOnlineUrl}"]`)
+                        .trigger('mouseover')
+
+                    cy.get(`[data-cy^="floating-button-has-error-${url}"]`)
                         .should('be.visible')
-                        .contains('Invalid file')
+                        .contains(errorMessage)
+
+                    // Mouseout to remove tooltip
+                    cy.wrap($layer).trigger('mouseout', { force: true })
                 } else {
-                    cy.get(`[data-cy^="floating-button-has-error-${outOfBoundKMLUrl}"]`)
-                        .should('be.visible')
-                        .contains('out of projection bounds')
+                    cy.get(`[data-cy^="floating-button-has-error-${url}"]`).should('not.exist')
                 }
+
+                // Ensure no spinner is shown
                 cy.wrap($layer)
                     .find('[data-cy^="button-loading-metadata-spinner"]')
                     .should('not.exist')
             })
 
         //---------------------------------------------------------------------
         // Test removing a layer
-        cy.log('Test removing all invalid kml layer')
-        cy.get(`[data-cy^="button-remove-layer-${invalidFileOnlineUrl}"]:visible`).click({
+        cy.log('Test removing all kml layer')
+        cy.get(
+            `[data-cy^="button-remove-layer-${validOnlineUrlWithInvalidContentType}-"]:visible`
+        ).click({
+            force: true,
+        })
+        cy.get(`[data-cy^="button-remove-layer-${invalidFileOnlineUrl}-"]:visible`).click({
```

Comment:
why is there a trailing `-` at the end here?
If the HTML is rendered with that character, I'd prefer we remove it, it's not "nice" 😉